ecs-init/cache/cache.go

```diff
@@ -34,16 +36,29 @@ const (
 
 // Downloader is resposible for cache operations relating to downloading the agent
 type Downloader struct {
-	getter httpGetter
-	fs     fileSystem
+	getter   httpGetter
+	fs       fileSystem
+	metadata instanceMetadata
+
+	region string
 }
 
 // NewDownloader returns a Downloader with default dependencies
-func NewDownloader() *Downloader {
-	return &Downloader{
+func NewDownloader() (downloader *Downloader) {
+	downloader = &Downloader{
 		getter: customGetter,
 		fs:     &standardFS{},
 	}
+
+	// NewSession can fail. If metadata cannot be initialized the region string is populated with the default value
+	// to prevent future calls to retrieve the region from metadata
+	sessionInstance, err := session.NewSession()
+	if err != nil {
+		downloader.region = config.DefaultRegionName
+	} else {
+		downloader.metadata = ec2metadata.New(sessionInstance)
+	}
+	return
```

Comment:
Please delete this line, or move it to inside the `if err != nil {` block so that you can get rid of `else`